src/main/scala/is/hail/expr/Relational.scala

```diff
@@ -1402,7 +1402,13 @@ case class MatrixMapCols(child: MatrixIR, newCol: IR, newKey: Option[IndexedSeq[
     val colValuesType = TArray(prev.typ.colType)
     val vaType = prev.typ.rvRowType
 
+    var initOpNeedsSA = false
+    var initOpNeedsGlobals = false
+
     val transformInitOp: (Int, IR) => IR = { (nAggs, initOpIR) =>
+      initOpNeedsSA = Mentions(initOpIR, "sa")
```

Comment:
I think you should break this out into a separate PR. It's a good change that would be nice to have irrespective of BGEN.